File: 1101-1200/1151-Minimum Swaps to Group All 1's Together/1151-Minimum Swaps to Group All 1's Together.cpp
class Solution {
public:
    int minSwaps(vector<int>& data) {
        int target = accumulate(data.begin(), data.end(), 0);
        int count = 0, minS = data.size();
        for (int i = 0; i < data.size(); ++i) {
            if (data[i]) {
                ++count;
            }
            if (i >= target - 1) {
                if (i >= target && data[i - target] == 1) {
                    --count;
                }
                minS = min(minS, target - count);
            }
        }
        return minS;
    }
};

File: 1101-1200/1151-Minimum Swaps to Group All 1's Together/1151-Minimum Swaps to Group All 1's Together.py
class Solution:
    def minSwaps(self, data: List[int]) -> int:
        target = data.count(1)
        count = 0
        minS = len(data)
        for i in range(len(data)):
            if data[i] == 1:
                count += 1
            if i >= target - 1:
                if i >= target and data[i - target] == 1:
                    count -= 1
                minS = min(minS, target - count)
        return minS
